src/app/page-courses/page-courses.component.ts

```diff
@@ -12,22 +14,12 @@ import {getLocaleDateFormat} from "@angular/common";
 export class PageCoursesComponent implements OnInit {
 	listCourses: Course[] = [];
 
-	constructor() {
-		this.listCourses.push(new Course({
-			id: 1,
-			title: 'demo Title',
-			creationDate: '10-10-2019',
-			duration: 15,
-			description: 'demo Description',
-		}));
-
-		this.listCourses.push(new Course({
-			id: 2,
-			title: 'demo Title 2',
-			creationDate: '10-10-2019',
-			duration: 15,
-			description: 'demo Description 2',
-		}));
+	constructor(coursesService: CoursesService) {
+		setTimeout(() => {
+			coursesService.getList().then((items: ICourse[]) => {
+				map(items, (item: CourseProps) => this.listCourses.push(new Course(item)));
```

Comment:
I suggest modigying coursesService to return Cource objects instead of raw objects with ICource interface.

Reply:
be later